util/io/src/main/scala/sbt/IO.scala

```diff
@@ -36,39 +36,43 @@ object IO {
   val utf8 = Charset.forName("UTF-8")
 
   /**
-   * Returns a URL for the directory or jar containing the the class file `cl`.
+   * Returns a URL for the classfile containing the given class
    * If the location cannot be determined, an error is generated.
    */
-  def classLocation(cl: Class[_]): URL = {
-    val codeSource = cl.getProtectionDomain.getCodeSource
-    if (codeSource ne null) {
-      codeSource.getLocation
-    } else {
-      // NB: This assumes that classes without code sources are System classes, and thus located in
-      // jars. It assumes that `urlAsFile` will truncate to the containing jar file.
-      val clsfile = s"${cl.getName.replace('.', '/')}.class"
-      Option(ClassLoader.getSystemClassLoader.getResource(clsfile))
-        .flatMap {
-          urlAsFile
-        }.getOrElse {
-          sys.error("No class location for " + cl)
-        }.toURI.toURL
+  def classfileLocation(cl: Class[_]): URL = {
```

Comment:
@eed3si9n Do we want to do something about this binary compatibility breakage before cutting a 0.13.10 release?